src/main/java/com/bio4j/model/NCBITaxonomyGraph.java

```diff
@@ -65,46 +67,66 @@
 
       Values are, among others, *species*, *genus*, or *no rank*; see the enum below.
     */
-    public final TaxonomicRank taxonomicRank = new TaxonomicRank();
-    public final class TaxonomicRank extends Property<TaxonomicRanks> implements FromAny, ToOne {
+    public final Rank rank = new Rank();
+    public final class Rank extends Property<TaxonomicRank> implements FromAny, ToOne {
 
-      private TaxonomicRank() { super(TaxonomicRanks.class); }
+      private Rank() { super(TaxonomicRank.class); }
     }
   }
 
   /*
     The set of valid ranks is nowhere documented; this is just an approximation.
   */
-  public static enum TaxonomicRanks {
-
-    noRank,
-    superkingdom,
-    kingdom,
-    superphylum,
-    phylum,
-    subphylum,
-    clazz, // reserved word
-    subclass,
-    superclass,
-    infraclass,
-    order,
-    parvorder,
-    suborder,
-    infraorder,
-    family,
-    subfamily,
-    superfamily,
-    tribe,
-    subtribe,
-    genus,
-    subgenus,
-    speciesGroup,
-    speciesSubgroup,
-    species,
-    subspecies,
-    varietas,
-    forma,
-    thereIsAnIndeterminateNumberOfRanksAsATaxonomistMayInventANewRankAtWillAtAnyTimeIfTheyFeelThisIsNecessary;
+  public static enum TaxonomicRank {
+
+    Superkingdom,
+    Kingdom,
+    Superphylum,
+    Phylum,
+    Subphylum,
+    Class,
+    Subclass,
+    Superclass,
+    Infraclass,
+    Order,
+    Parvorder,
+    Suborder,
+    Infraorder,
+    Family,
+    Subfamily,
+    Superfamily,
+    Tribe,
+    Subtribe,
+    Genus,
+    Subgenus,
+    SpeciesGroup,
+    SpeciesSubgroup,
+    Species,
+    Subspecies,
+    Varietas,
+    Forma,
+    // "no rank" is a valid value
+    NoRank,
+    // NOTE: there is an indeterminate number of ranks as a taxonomist may invent a new rank at will at any time if they feel this is necessary; we reserve a special value for such cases:
+    UNKNOWN;
+
+
+    /* Converts values of the enum to lower-case strings, adds a space for NoRank */
+    @Override public String toString() {
+      switch(this) {
+        case NoRank:  return "no rank";
+        case UNKNOWN: return "UNKNOWN";
+        default:      return this.name().toLowerCase(Locale.ENGLISH);
+      }
+    }
+
+    /* Converts strings to enum values _ignoring case_; Uses UNKNOWN as a default value */
+    public static TaxonomicRank fromString(String name) {
+      return Arrays.stream(TaxonomicRank.values())
+        .filter(rank -> rank.toString().equalsIgnoreCase(name))
+        .findFirst()
+        .orElse(UNKNOWN);
```

Comment:
This will return `TaxonomicRank.UNKNOWN` not only for any unexpected string, but also for `null` input. Should it throw `NullPointerException` instead? (as `Enum.valueOf()` does)